FLT/AutomorphicForm/QuaternionAlgebra/HeckeOperators/Local.lean

```diff
@@ -0,0 +1,341 @@
+/-
+Copyright (c) 2025 Bryan Wang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard, Bryan Wang
+-/
+import FLT.AutomorphicForm.QuaternionAlgebra.HeckeOperators.Abstract -- abstract Hecke ops
+import FLT.AutomorphicForm.QuaternionAlgebra.Defs -- definitions of automorphic forms
+import FLT.QuaternionAlgebra.NumberField -- rigidifications of quat algs
+import FLT.DedekindDomain.FiniteAdeleRing.LocalUnits -- for (π 0; 0 1)
+import Mathlib.Data.Matrix.Reflection
+
+open NumberField IsQuaternionAlgebra.NumberField IsDedekindDomain
+open TotallyDefiniteQuaternionAlgebra
+open IsDedekindDomain.HeightOneSpectrum
+open scoped TensorProduct
+open scoped Pointwise
+
+namespace TotallyDefiniteQuaternionAlgebra.WeightTwoAutomorphicForm.HeckeOperator
+
+-- let F be a (totally real) number field
+variable {F : Type*} [Field F] [NumberField F]
+
+namespace Local
+
+variable {v : HeightOneSpectrum (𝓞 F)}
+
+variable (α : v.adicCompletionIntegers F)
+
+variable (hα : α ≠ 0)
+
+-- Not sure if this could be somewhere else.
+variable (v) {α hα} in
+lemma valuation_lt_one_iff_mem_maximalIdeal {t : adicCompletionIntegers F v} :
+    Valued.v (t : adicCompletion F v) < 1
+    ↔ t ∈ IsLocalRing.maximalIdeal (adicCompletionIntegers F v) :=
+  Iff.trans
+  (Valuation.isEquiv_iff_val_lt_one.mp
+    (Valuation.isEquiv_valuation_valuationSubring Valued.v))
+  (ValuationSubring.valuation_lt_one_iff (adicCompletionIntegers F v) _).symm
```

Comment:
Just to let you know: you can do this
```suggestion
  (Valuation.isEquiv_iff_val_lt_one.mp
    (Valuation.isEquiv_valuation_valuationSubring Valued.v)).trans
  (ValuationSubring.valuation_lt_one_iff (adicCompletionIntegers F v) _).symm
```
"Golfing" is encouraged in mathlib (the logic seems to be "the shorter the code, the faster it is to compile and the fewer bytes it takes") but I'm less bothered about it here (and I'm not entirely sure I buy the logic).